### PR TITLE
[Beam 287] Add new groupId

### PR DESCRIPTION
**Pull request by jbonofre**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [X] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [X] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [X] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [X] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---
Add new Beam groupId.

### CODE REVIEW

**Comment by jbonofre** [comment]
R: @davorbonaci

File: examples/java/pom.xml [hunk 1]
```diff
@@ -214,7 +214,7 @@
     </dependency>
 
     <dependency>
-      <groupId>org.apache.beam</groupId>
+      <groupId>org.apache.beam.runners</groupId>
```
Comment (davorbonaci) [inline examples/java/pom.xml]: google-cloud-dataflow-java (i.e., drop runner from the artifact id)

File: pom.xml [hunk 2]
```diff
@@ -245,13 +245,13 @@
       </dependency>
 
       <dependency>
-        <groupId>org.apache.beam</groupId>
+        <groupId>org.apache.beam.runners</groupId>
         <artifactId>runners-core</artifactId>
```
Comment (davorbonaci) [inline pom.xml]: `core` (i.e., drop runners prefix)

File: runners/core-java/pom.xml [hunk 3]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.runners</groupId>
     <artifactId>runners-parent</artifactId>
```
Comment (davorbonaci) [inline runners/core-java/pom.xml]: `parent`, i.e. drop runners prefix

File: runners/flink/examples/pom.xml [hunk 4]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.runners</groupId>
     <artifactId>flink-runner-parent</artifactId>
```
Comment (davorbonaci) [inline runners/flink/examples/pom.xml]: `flink-parent`

File: runners/flink/examples/pom.xml [hunk 5]
```diff
@@ -70,7 +70,7 @@
   <dependencies>
 
     <dependency>
-      <groupId>org.apache.beam</groupId>
+      <groupId>org.apache.beam.runners</groupId>
       <artifactId>flink-runner_2.10</artifactId>
```
Comment (davorbonaci) [inline runners/flink/examples/pom.xml]: flink_2.10

File: runners/flink/pom.xml [hunk 6]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.runners</groupId>
     <artifactId>runners-parent</artifactId>
```
Comment (davorbonaci) [inline runners/flink/pom.xml]: parent

File: sdks/java/extensions/join-library/pom.xml [hunk 7]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.extensions</groupId>
```
Comment (davorbonaci) [inline sdks/java/extensions/join-library/pom.xml]: parent

File: sdks/java/io/google-cloud-platform/pom.xml [hunk 8]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.io</groupId>
     <artifactId>io-parent</artifactId>
```
Comment (davorbonaci) [inline sdks/java/io/google-cloud-platform/pom.xml]: parent

File: sdks/java/io/hdfs/pom.xml [hunk 9]
```diff
@@ -22,7 +22,7 @@
   <modelVersion>4.0.0</modelVersion>
 
   <parent>
-    <groupId>org.apache.beam</groupId>
+    <groupId>org.apache.beam.io</groupId>
     <artifactId>io-parent</artifactId>
```
Comment (davorbonaci) [inline sdks/java/io/hdfs/pom.xml]: parent

**Comment by davorbonaci** [comment]
Left a few minor comments, almost ready for merge.

**Comment by davorbonaci** [comment]
Triage: 0.1.0-incubating, release blocking.

**Comment by jbonofre** [comment]
Updated accordingly.

**Comment by jbonofre** [comment]
Squashed 3 commits in one.

File: runners/flink/examples/pom.xml [hunk 10]
```diff
@@ -70,8 +70,8 @@
   <dependencies>
 
     <dependency>
-      <groupId>org.apache.beam</groupId>
-      <artifactId>flink-runner_2.10</artifactId>
+      <groupId>org.apache.beam.runners</groupId>
+      <artifactId>flink-_2.10</artifactId>
```
Comment (davorbonaci) [inline runners/flink/examples/pom.xml]: typo here

File: runners/flink/examples/pom.xml [hunk 11]
```diff
@@ -22,13 +22,13 @@
...
-    <groupId>org.apache.beam</groupId>
-    <artifactId>flink-runner-parent</artifactId>
+    <groupId>org.apache.beam.runners</groupId>
+    <artifactId>flink-parent</artifactId>
     <version>0.1.0-incubating-SNAPSHOT</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
 
-  <artifactId>flink-runner-examples_2.10</artifactId>
+  <artifactId>flink-examples_2.10</artifactId>
```
Comment (davorbonaci) [inline runners/flink/examples/pom.xml]: flink 2.10 examples?

File: runners/flink/runner/pom.xml [hunk 12]
```diff
@@ -22,13 +22,13 @@
...
-    <groupId>org.apache.beam</groupId>
-    <artifactId>flink-runner-parent</artifactId>
+    <groupId>org.apache.beam.runners</groupId>
+    <artifactId>flink-parent</artifactId>
     <version>0.1.0-incubating-SNAPSHOT</version>
     <relativePath>../pom.xml</relativePath>
   </parent>
 
-  <artifactId>flink-runner_2.10</artifactId>
+  <artifactId>flink-_2.10</artifactId>
```
Comment (davorbonaci) [inline runners/flink/runner/pom.xml]: typo

File: runners/flink/examples/pom.xml (hunk 10 above)
Comment (jbonofre) [inline runners/flink/examples/pom.xml]: Right, fixing.

**Comment by davorbonaci** [comment]
LGTM, modulo 2 typos.

**Comment by jbonofre** [comment]
Updated fixing typo.

**Comment by davorbonaci** [comment]
LGTM. Waiting for CI to complete.

**Comment by jbonofre** [comment]
Travis failure comes from Kafka, not from the PR itself.

**Comment by davorbonaci** [comment]
Merging

**Pull request closed without merging**